### DISCUSSION TITLE
React hook form  controller

### DISCUSSION THREAD

**Discussion by karthiktrp102** [react-hook-form/react-hook-form#5491] [General] [answered]
here I am using React hook form for validation I need to add validations for either first name or last name suppose if I only fill the first name there is no need to fill the last name so I am able to submit or if I only fill the last name there is no need to fill the first name so I need to validate like that can anyone help me
![1](https://user-images.githubusercontent.com/77161149/120804702-9871e500-c562-11eb-978e-632f9cc9d12d.jpg)
![2](https://user-images.githubusercontent.com/77161149/120804716-9b6cd580-c562-11eb-8c43-cc2026b35cde.jpg)

**Comment by thanh-nguyen-95** [comment, marked as answer]
First of all, that's not how you use `Controller` for custom input. Please read the documentation again:

- https://react-hook-form.com/get-started#IntegratingControlledInputs
- https://react-hook-form.com/api/useform/register

For you needs, you can use `validate` and `getValues` to validate each field. Some thing like this:

```jsx
function isFirstNameOrLastNameFilled() {
  const firstName= getValues('firstName')
  const lastName = getValues('lastName')

  const isValid = firstName || lastName

  if (!isValid) return 'Your error message here'
  return true
}

<Controller
  name="firstName"
  rules={{ validate: isFirstNameOrLastNameFilled }}
  render={ /* render the input here */ }
/>
```

**Reply by bluebill1049** [reply]
Thank you @thanh-nguyen-95
